File: src/exemplos/5_EstruturasDeDados/4_Vetor/2-vetor.py
#  -*- coding: utf-8 -*-
#       @file: 2-vetor.py
#     @author: Guilherme N. Ramos (gnramos@unb.br)
# @disciplina: Algoritmos e Programação de Computadores
#
# Exemplo uso de vetor como argumento de função.


def mostra(vetor):
    """Mostra todos os elementos do vetor."""
    print "{", ' '.join(str(v) for v in vetor), '}'


def maior(vetor):
    """Retorna o maior elemento do vetor."""
    return max(vetor)

    # m = 0
    # for i in range(len(vetor)):
    #     if vetor[i] > vetor[m]:
    #         m = i
    # return vetor[m]
    #
    # ou então
    #
    # m = vetor[0]
    # for v in vetor:
    #     if v > m:
    #         m = v
    # return m


vetor = [1, 2, 1, 4, 8, 7, 6, 5, 10, 9]
mostra(vetor)
print "O maior elemento é:", maior(vetor)

File: src/exemplos/5_EstruturasDeDados/4_Vetor/2-vetor.c
/**      @file: 2-vetor.c
 *     @author: Guilherme N. Ramos (gnramos@unb.br)
 * @disciplina: Algoritmos e Programação de Computadores
 *
 * Exemplo uso de vetor como argumento de função. */

#include "apc_vetor.h"

#define TAM 10

/* Retorna o maior elemento do vetor. */
int maior(int vetor[TAM]) {
/* O tipo é "vetor de TAM elementos". */
  int i, maior = 0;

  for(i = 1; i < TAM; ++i)
    if(vetor[i] > vetor[maior])
        maior = vetor[i];

  return maior;
}

int main() {
  int vetor[] = {1,2,1,4,8,7,6,5,10,9};

  mostra_i(vetor, TAM);
  printf("O maior elemento é: %d\n", maior(vetor));

  return 0;
}